certmanager/util.go

```diff
@@ -18,10 +18,43 @@ import (
 	"crypto"
 	"crypto/x509"
 	"encoding/pem"
+	"net/http"
 
+	// Install this package by $ go get -u cloud.google.com/go/...
+	"cloud.google.com/go/compute/metadata"
 	"github.com/golang/glog"
 )
 
+type TokenFetcher interface {
+	FetchToken() (string, error)
+}
+
+type GcpTokenFetcher struct {
+	// aud is the unique URI agreed upon by both the instance and the system verifying the instance's identity.
+	// For more info: https://cloud.google.com/compute/docs/instances/verifying-instance-identity
+	aud            string
+	serviceAccount string
+}
+
+func (fetcher GcpTokenFetcher) setAudience(audience string) {
+	fetcher.aud = audience
+}
+
+func (fetcher GcpTokenFetcher) setServiceAccount(sa string) {
+	fetcher.serviceAccount = sa
+}
+
+func (fetcher GcpTokenFetcher) getTokenUri() string {
+	// The GCE metadata service URI to get identity token.
+	return "instance/" + fetcher.serviceAccount + "/default/identity?audience=" + fetcher.aud
```

Comment:
Sorry for raising this again (and let me know if I've misunderstood what you're trying to do).

According to https://cloud.google.com/compute/docs/storing-retrieving-metadata, the URL for a service account named `foo` should be `blah/instance/service-accounts/foo/identity?...`. However, here the uri is constructed as `blah/instance/foo/default/identity?...`. Can you double-check which one is correct?